src/schema_registry/telemetry/middleware.py

```diff
@@ -20,12 +23,59 @@ async def telemetry_middleware(
     request: Request,
     call_next: Callable[[Request], Awaitable[Response]],
     tracer: Tracer = Provide[TelemetryContainer.tracer],
+    meter: Meter = Provide[TelemetryContainer.meter],
 ) -> Response:
     resource = request.url.path.split("/")[1]
     with tracer.get_tracer().start_as_current_span(name=f"{request.method}: /{resource}", kind=SpanKind.SERVER) as span:
+        span.add_event("Creating metering resources")
+        karapace_http_requests_in_progress: UpDownCounter = meter.get_meter().create_up_down_counter(
+            name="karapace_http_requests_in_progress",
+            description="In-progress requests for HTTP/TCP Protocol",
+        )
+        karapace_http_requests_duration_seconds: Histogram = meter.get_meter().create_histogram(
+            unit="seconds",
+            name="karapace_http_requests_duration_seconds",
+            description="Request Duration for HTTP/TCP Protocol",
+        )
+        karapace_http_requests_total: Counter = meter.get_meter().create_counter(
+            name="karapace_http_requests_total",
+            description="Total Request Count for HTTP/TCP Protocol",
+        )
+
+        # Set start time for request
+        setattr(request.state, meter.START_TIME_KEY, time.monotonic())
+
+        # Extract request labels
+        path = request.url.path
```

Comment:
This will create quite large cardinality when Karapace handles large number of topics and versions.

Reply:
I will use the base resource name, i.e. `POST: /subjects/xyz` will instrument `/subjects`, this is the only way I can think of now, aside the custom `APIRouter`.